src/lib/reasoners/uf.ml

```diff
@@ -1107,9 +1213,38 @@ module MED = Map.Make
         else Expr.compare a b
     end)
 
-let terms env = ME.fold MED.add env.make MED.empty
+let is_suspicious_name hs =
+  match Hstring.view hs with
+  | "@/" | "@%" | "@*" -> true
+  | _ -> false
+
+(* The model generation is known to be imcomplete for FPA and Bitvector
+   theories. *)
+let is_suspicious_symbol = function
+  | Sy.Op (Float | Fixed | Abs_int | Abs_real | Sqrt_real
+          | Sqrt_real_default | Sqrt_real_excess
+          | Real_of_int | Int_floor | Int_ceil
+          | Max_int | Max_real | Min_int | Min_real
+          | Pow | Integer_log2 | Int2BV _ | BV2Nat
+          | BVand | BVor | Integer_round) -> true
```

Comment:
After #886, `BV2Nat`, `Int2BV` and `Pow` should no longer be treated as suspicious. Further, `Float`, `Abs_int`, `Abs_real`, `Sqrt_real_default`, `Sqrt_real_excess`, `Real_of_int`, `Int_floor`, `Int_ceil`, `Max_int`, `Max_real`, `Min_int`, `Min_real`, `Integer_log2` and `Integer_round` are now only suspicious if the FPA prelude is not loaded.

This is because we now propagate their arguments during model generation, which should prevent us from generating models with invalid values for these functions.